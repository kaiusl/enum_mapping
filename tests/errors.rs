use enum_maping::EnumMaping;


#[derive(EnumMaping, Debug, Eq, PartialEq)]
enum Example {
    #[mapstr(name = "vname", "variant_1")]
    #[mapstr(name = "short", "V1", default_to="unknown", default_from="Unknown")]
    #[mapstr(name = "pretty_vname", "Variant 1")]
    #[mapstr(name = "caps", "VARIANT_1", to=false)]
    V1,

    #[mapstr("variant_2")]
    #[mapstr("V2")]
    #[mapstr("Variant 2")]
    #[mapstr(name = "caps2", "VARIANT_1", from=false)]
    V2,

    #[mapstr(name = "pretty_vname", "Variant 3")]
    #[mapstr("V3")]
    #[mapstr(name = "vname", "variant_3")]
    V3,

    #[mapstr(name = "vname", default, "unknown")]
    Unknown,

    #[mapstr(name = "error", "err")]
    Error
}

fn main() {
    Example::V1.try_to_caps(); 
    Example::V2.try_from_caps2(); 
} 

#[derive(EnumMaping)]
enum Ex0 {
    #[mapstr()]
    V1,
}

#[derive(EnumMaping)]
enum Ex {
    #[mapstr("fa")]
    V1,
}

#[derive(EnumMaping)]
enum Ex2 {
    #[mapstr(name="fas")]
    V1,
}

#[derive(EnumMaping)]
enum Ex3 {
    #[mapstr(name="n", "fa", "faas")]
    V1,
}


#[derive(EnumMaping)]
enum Ex4 {
    #[mapstr(name="n", "fa", fas)]
    V1,
}

#[derive(EnumMaping)]
enum Ex5 {
    #[mapstr(name="n", "fa", nasdf="faas")]
    V1,
}

#[derive(EnumMaping)]
enum Ex6 {
    #[mapstr(name="n", "fa", name=true)]
    V1,
}

#[derive(EnumMaping)]
enum Ex7 {
    #[mapstr2(name="n", "fa", name=true)]
    V1,
}



#[derive(EnumMaping)]
struct Sa {

}