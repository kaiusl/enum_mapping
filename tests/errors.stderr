error: Missing a value to map to. Simplest form should be #[mapstr(name="name", "value")]
  --> tests/no_to.rs:35:10
   |
35 | #[derive(EnumMaping)]
   |          ^^^^^^^^^^
   |
   = note: this error originates in the derive macro `EnumMaping` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Parameter `name` must be specified. Simplest form should be #[mapstr(name="name", "value")]
  --> tests/no_to.rs:43:14
   |
43 |     #[mapstr("fa")]
   |              ^^^^

error: Missing a value to map to. Simplest form should be #[mapstr(name="name", "value")]
  --> tests/no_to.rs:49:14
   |
49 |     #[mapstr(name="fas")]
   |              ^^^^^^^^^^

error: `value` is set twice. Remove one.
  --> tests/no_to.rs:55:30
   |
55 |     #[mapstr(name="n", "fa", "faas")]
   |                              ^^^^^^

error: Found unknown `mapstr` parameter
  --> tests/no_to.rs:62:30
   |
62 |     #[mapstr(name="n", "fa", fas)]
   |                              ^^^

error: Found unknown `mapstr` parameter.
  --> tests/no_to.rs:68:30
   |
68 |     #[mapstr(name="n", "fa", nasdf="faas")]
   |                              ^^^^^^^^^^^^

error: Found unknown `mapstr` parameter.
  --> tests/no_to.rs:74:30
   |
74 |     #[mapstr(name="n", "fa", name=true)]
   |                              ^^^^^^^^^

error: Found unknown attribute on variant.
  --> tests/no_to.rs:80:7
   |
80 |     #[mapstr2(name="n", "fa", name=true)]
   |       ^^^^^^^

error: Derive macro `EnumMaping` expected an enum.
  --> tests/no_to.rs:87:1
   |
87 | / struct Sa {
88 | |
89 | | }
   | |_^

error: cannot find attribute `mapstr2` in this scope
  --> tests/no_to.rs:80:7
   |
80 |     #[mapstr2(name="n", "fa", name=true)]
   |       ^^^^^^^ help: a derive helper attribute with a similar name exists: `mapstr`

error[E0599]: no method named `try_to_caps` found for enum `Example` in the current scope
  --> tests/no_to.rs:31:17
   |
5  | enum Example {
   | ------------ method `try_to_caps` not found for this
...
31 |     Example::V1.try_to_caps();
   |                 ^^^^^^^^^^^ help: there is an associated function with a similar name: `try_to_caps2`

error[E0599]: no method named `try_from_caps2` found for enum `Example` in the current scope
  --> tests/no_to.rs:32:17
   |
5  | enum Example {
   | ------------ method `try_from_caps2` not found for this
...
32 |     Example::V2.try_from_caps2();
   |                 ^^^^^^^^^^^^^^ help: there is an associated function with a similar name: `try_to_caps2`
